test/rule-matches/aria-form-field-label-matches.js

```diff
@@ -0,0 +1,91 @@
+describe('aria-form-field-label-matches', function() {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var queryFixture = axe.testUtils.queryFixture;
+	var rule = axe._audit.rules.find(function(rule) {
+		return rule.id === 'aria-form-field-label';
+	});
+
+	afterEach(function() {
+		fixture.innerHTML = '';
+	});
+
+	it('returns false when node has no role', function() {
+		var vNode = queryFixture('<textarea id="target" title="Label"></textarea>');
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns false for node `map area[href]`', function() {
+		var vNode = queryFixture(
+			'<map><area id="target" href="#" role="randomRole"></map>'
```

Comment:
`randomRole` would return false anyway. This test doesn't prove what you're trying to prove.